src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,185 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> after = node(index);
+            Node<T> before = after.prev;
+            Node<T> newNode = new Node<>(before, value, after);
+            after.prev = newNode;
+            if (before == null) {
+                head = newNode;
+            } else {
+                before.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return node(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node = node(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> i = head; i != null; i = i.next) {
```

Comment:
Let's rename `i`, cause it is not an index. 
And I think will be better to use `while` for iterating, because `for with index` is commonly used for iterating structures that have index, which is absent here.